### PR TITLE
Allow reconfiguring several configuration fields to 0 (default)

### PR DESCRIPTION
**Pull request by jeffreylovitz**
Previously, GRAPH.CONFIG SET erroneously would not allow the timeout to be returned to 0 (unlimited), as it would only accept positive integers.

This PR removes that limitation by accepting any non-negative integer for the timeout value. It additionally resolves the same bug for `DELTA_MAX_PENDING_CHANGES` and `QUERY_MEM_CAPACITY`.

### CODE REVIEW

File: tests/flow/test_config.py [hunk 1]
```diff
@@ -144,3 +144,29 @@ def test07_config_invalid_subcommand(self):
             assert("Unknown subcommand for GRAPH.CONFIG" in str(e))
             pass
 
+    def test08_config_restore_timeout_to_default(self):
```
Comment (swilly22) [inline tests/flow/test_config.py:147]: Please rename or break, as this test updates `QUERY_MEM_CAPACITY` as well.

File: tests/flow/test_config.py [hunk 2]
```diff
@@ -144,3 +144,29 @@ def test07_config_invalid_subcommand(self):
...
+        self.env.assertEqual(response, "OK")
+
+        # Make sure config been updated.
+        response = redis_con.execute_command("GRAPH.CONFIG GET TIMEOUT")
+        expected_response = ["TIMEOUT", 0]
+        self.env.assertEqual(response, expected_response)
+
+        # Issue long-running query to validate the reconfiguration
+        result = redis_graph.query("UNWIND range(1,1000000) AS v RETURN COUNT(v)")
+        self.env.assertEqual(result.result_set[0][0], 1000000)
```
Comment (swilly22) [inline tests/flow/test_config.py:172]: Make sure this query timeouts prior to setting timeout to unlimited.

File: src/configuration/config.c [hunk 3]
```diff
@@ -615,7 +623,7 @@ bool Config_Option_set(Config_Option_Field field, const char *val) {
 		case Config_TIMEOUT:
 			{
 				long long timeout;
-				if(!_Config_ParsePositiveInteger(val, &timeout)) return false;
+				if(!_Config_ParseNonNegativeInteger(val, &timeout)) return false;
 				Config_timeout_set(timeout);
 			}
 			break;
```
Comment (swilly22) [inline src/configuration/config.c:629]: `Config_RESULTSET_MAX_SIZE` should use `_Config_ParsePositiveInteger`
`Config_VKEY_MAX_ENTITY_COUNT` should use `_Config_ParseNonNegativeInteger`

**Review by swilly22 — APPROVED** [review]
@jeffreylovitz please see additional comments.

File: src/configuration/config.c (hunk 3 above)
Comment (jeffreylovitz) [reply src/configuration/config.c:629]: `Config_RESULTSET_MAX_SIZE` defaults to unlimited when given a negative value, so I think `_Config_ParseInteger` makes sense for it.

**Pull request merged**